### DISCUSSION TITLE
pyLossless on HPC (Digital Research Alliance of Canada )

### DISCUSSION THREAD

**Discussion by scott-huberty** [lina-usc/pylossless#126] [General]
I have EEG files that will run through pylossless in < 30 minutes each on my local computer (16 GB desktop), where for example, FASTICA on average finishes in 3 minutes. 

These same files on the DRAC HPC (I am using Graham), run just as fast through the pipeline methods UP to ICA. But ICA is a total bottleneck.

On the HPC (Graham), on average my files get to the first ICA (FASTICA) step within a few minutes of pylossless starting. FASTICA won't finish by the time the SLURM job times out (2-hour job, 16 GB memory - which should be way over kill...). 

@Andesha I know you battled with this with your files too. I remember you describing that `mne` / `sklearn` use `jobllib` in the backend, which doesn't recognize the available CPUs etc in the HPC environment.
 
It would be great to be able to run pylossless on the HPC clusters. Do you have any suggestions?

**Comment by scott-huberty** [comment]
In summary: FastICA takes 3 minutes on my computer. On the HPC, FastICA won't finish, even after running for _hours_ (record job length so far: 8 hours...).

**Comment by christian-oreilly** [comment]
Could it be due to how the packages are built (e.g., NumPy)? Some numerical routines can be compiled in different ways (e.g., BLAS and LAPACK) with a significant impact on processing time if I remember well...

**Comment by Andesha** [comment]
I’m away for a bit but you’re exactly right. This was part of what I ran
into. Hopefully Monday I can hand you my recipe for getting things
installed properly and it’ll run no problem.
